bitcoin/src/crypto/ecdsa.rs

```diff
@@ -293,6 +295,41 @@ impl From<DecodeError> for ParseSignatureError {
     fn from(e: DecodeError) -> Self { Self::Decode(e) }
 }
 
+#[cfg(feature = "arbitrary")]
+impl<'a> Arbitrary<'a> for Signature {
+    fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
+        // The valid range of r and s should be between 1 and n-1 where
+        // n = 0xFFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFE BAAEDCE6 AF48A03B BFD25E8C D0364141
+        let high_min = 0x0u128;
+        let high_max = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEu128;
+        let low_min = 0x1u128;
```

Comment:
I'm way out of my depth on cryptography related things, so I did some light research before putting this together. Just for my information, why is zero allowed? The material that i read led me to believe that `r` and `s` should be in the range of [1, n-1], and I also noticed that `Signature::from_compact` was not giving me an error when I used zero for those values as well. Thanks

Reply:
For whatever reason, 0 in libsecp256k1 zero a "validation error" rather than a "parse error". The distinction is somewhat artificial -- as Bitcoin Core uses it, signatures are only parsed right before validation, so both kinds of errors are treated the same.

Given that out-of-range values are parse errors, and a 0 signature is similarly easy to detect and guaranteed to fail, it may make sense to file a bug upstream about this.

Reply:
Okay, I think it makes sense if I change the minimum bound to zero, and then file the bug. If/when it gets approved, the tests will fail and then we can change it to one